File: Sheet #2 (Loops)/I_Palindrome.cpp
#include <bits/stdc++.h>

using namespace std;
int main()
{
    int n, rev = 0, temp;
    cin >> n;

    int num = n;
    do
    {
        temp = num % 10;
        rev = rev * 10 + temp;
        num /= 10;
    } while (num != 0);
    cout << rev << "\n";

    if (rev != n)
        cout << "NO";
    else
        cout << "YES";

    return 0;
}

File: Sheet #2 (Loops)/I_Palindrome.java
import java.util.*;

public class I_Palindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int rev = 0, temp = n;
        while (n != 0) {
            int ld = n % 10;
            rev = rev * 10 + ld;
            n = n / 10;
        }
        System.out.println(rev);
        if (rev == temp)
            System.out.println("YES");
        else
            System.out.println("NO");
    }
}